ovs-p4rt/sidecar/ovsp4rt.cc

```diff
@@ -2201,62 +2203,49 @@ void ovsp4rt_config_fdb_entry(struct mac_learning_info learn_info,
       auto status_or_read_response =
           GetFdbTunnelTableEntry(session.get(), learn_info, p4info);
       if (status_or_read_response.ok()) {
-        printf("TUNNEL: read FDB L2_FWD_TX_TABLE entry present\n");
+        ovsp4rt_log_error(
+            "Error adding to FDB Tunnel Table: entry already exists");
         return;
       }
     }
 
     status = ConfigFdbTunnelTableEntry(session.get(), learn_info, p4info,
                                        insert_entry);
-    if (!status.ok())
-      printf(
-          "%s: Failed to program l2_fwd_tx_table for tunnel with mac "
-          "%x:%x:%x:%x:%x:%x\n",
-          insert_entry ? "ADD" : "DELETE", learn_info.mac_addr[0],
-          learn_info.mac_addr[1], learn_info.mac_addr[2],
-          learn_info.mac_addr[3], learn_info.mac_addr[4],
-          learn_info.mac_addr[5]);
+    if (!status.ok()) {
+      LogTableErrorWithMacAddr(insert_entry, "FDB Tunnel Table",
```

Comment:
We should leave all table names as they are defined earlier since that is what we are looking at during debugging. Table names such as `l2_fwd_tx_table` or `l2_tunnel_to_v4_table` that can be correlated while also looking at the P4 program file

Reply:
There are a number of problems with doing this:

1. It's not target/program-agnostic: it assumes that all P4 programs for all targets use the same table names for the same purposes, and will continue to do so in the future.
2. It bakes the table names into the code, in a section of the code that is separate from the places where the table names are defined and used.
3. It does not correlate with the software abstraction. The function `ConfigFdbTxVlanTableEntry()` maps to `L2_FWD_TX_TABLE`. 
4. Whom are the messages intended to benefit? The customer, or the developer?

Messages that are displayed by default (as these are) should be customer-friendly. This means using clear terminology and giving some indication of what the user is supposed to do in response to the message.

Messages that display P4 table names require inside knowledge. They are not customer friendly.

Error, warning, and info messages should be written with the customer in mind. Debug messages may be developer-focused.

I'm also concerned by the fact that the cavalier error handling. Failures during processing are sometimes logged and sometimes ignored. In most of the cases, processing continues after a failure has been detected. What state does this leave the remote system in? And you don't get a whole lot of information when you do get a log message. In some cases, there may several internal functions that update the same P4 table. Logging the table name with no additional information doesn't test you what kind of update failed. (That's one advantage of using the abstract name.) And except for the cases where a MAC address is displayed (something you recently expanded), there's no detail to help you understand which updates failed.

If I was participating in the design, I would request articulated philosophies/policies for both error handling and logging. And, if the policies didn't make it clear, I would ask for comments in the code explaining the decisions to act on some failures and ignore others.

"What we've got here is ... failure to communicate." _Cool Hand Luke_

Reply:
In any case, I was thinking of creating a bunch of `constexpr`s to define the text to be displayed, labeled with the abstract name, and putting them in a header file. This would handle any differences between DPDK and ES2K, and it would make it to change our minds over what to display.

Reply:
I might also create a DETAIL message class that can be used to enable additional information to be displayed with a DEBUG message.

Reply:
Please mull these thoughts over. We can then talk about how best to proceed.

Reply:
I've reengineered the logging code to display table names without violating the abstraction model.